airbyte-webapp/src/hooks/services/FormChangeTracker/hooks.ts

```diff
@@ -10,6 +10,8 @@ export const useUniqueFormId = (formId?: string) => useMemo(() => formId ?? uniq
 
 export const useFormChangeTrackerService = (): FormChangeTrackerServiceApi => {
   const [changedFormsById, setChangedFormsById] = useChangedFormsById();
+  const changedFormsByIdRef = useRef(changedFormsById);
+  changedFormsByIdRef.current = changedFormsById;
```

Comment:
Is this line actually necessary? The [docs](https://reactjs.org/docs/hooks-reference.html#useref) say:
> useRef returns a mutable ref object whose .current property is initialized to the passed argument (initialValue)

Reply:
Oh, is this necessary because the `changedFormsById` object will be modified from one render to the next, so this line updates the `current` pointer to the new object whenever it changes?

Reply:
> so this line updates the current pointer to the new object whenever it changes

Exactly, when the dirty state changes, the whole object gets replaced here:
https://github.com/airbytehq/airbyte/blob/b72cd7150ddce3caa5290953f90aec3822bf53aa/airbyte-webapp/src/hooks/services/FormChangeTracker/hooks.ts#L25

So in order to not bust the `useCallbacks` we need to ref it.